api/pkg/transformer/pipeline/upi_autoloading_op.go

```diff
@@ -0,0 +1,192 @@
+	}
+	return ua.autoloadingPostprocessInput(ctx, env)
+}
+
+func validateRequest(payload *types.UPIPredictionRequest) error {
+	if payload.PredictionTable != nil {
+		if err := validateUPITable(payload.PredictionTable); err != nil {
+			return err
+		}
+	}
+
+	if payload.TransformerInput == nil {
+		return nil
+	}
+
+	for _, tbl := range payload.TransformerInput.Tables {
+		if tbl == nil {
+			continue
+		}
+		if err := validateUPITable(tbl); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateUPITable(tbl *upiv1.Table) error {
+	// if user defined row_id in the columns then it will fail
+	// since row_id will be automated created from row entry
+	for _, col := range tbl.Columns {
+		if col.Name == table.RowIDColumn {
+			return fmt.Errorf("row_id column is reserved, user is not allowed to define explicitly row_id column")
+		}
+	}
+	return nil
+}
+
+func (ua *UPIAutoloadingOp) autoLoadingPreprocessInput(ctx context.Context, env *Environment) error {
+	requestPayload := env.symbolRegistry.RawRequest()
+	upiRequestPayload, valid := requestPayload.(*types.UPIPredictionRequest)
+	if !valid {
+		return fmt.Errorf("raw request is not valid")
+	}
+
+	if err := validateRequest(upiRequestPayload); err != nil {
+		return err
+	}
+
+	variableSymbols, err := generateVariableFromRequest(upiRequestPayload, env)
+	if err != nil {
+		return err
+	}
+	tableSymbols, err := generateTableFromRequest(upiRequestPayload, env)
+	if err != nil {
+		return err
+	}
+	mergedSymbols := mergeMap(variableSymbols, tableSymbols)
+	for name, val := range mergedSymbols {
+		env.SetSymbol(name, val)
+	}
+	if ua.OperationTracing != nil {
+		if err := ua.OperationTracing.AddInputOutput(nil, mergedSymbols); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func mergeMap(left, right map[string]any) map[string]any {
+	for k, v := range right {
+		left[k] = v
+	}
+	return left
+}
+
+func generateVariableFromRequest(requestPayload *types.UPIPredictionRequest, env *Environment) (map[string]any, error) {
+	variables := make(map[string]any)
+
+	if requestPayload.TransformerInput == nil {
+		return variables, nil
+	}
+
+	for _, variable := range requestPayload.TransformerInput.Variables {
+		switch variable.Type {
+		case upiv1.Type_TYPE_INTEGER:
+			variables[variable.Name] = variable.IntegerValue
+		case upiv1.Type_TYPE_DOUBLE:
+			variables[variable.Name] = variable.DoubleValue
+		case upiv1.Type_TYPE_STRING:
+			variables[variable.Name] = variable.StringValue
+		default:
+			return nil, fmt.Errorf("unknown type %T", variable)
+		}
+	}
+	return variables, nil
+}
+
+func generateTableFromRequest(requestPayload *types.UPIPredictionRequest, env *Environment) (map[string]any, error) {
+	tables := make(map[string]any)
+	if requestPayload.PredictionTable != nil {
+		predictionTable, err := table.NewFromUPITable(requestPayload.PredictionTable)
+		if err != nil {
+			return nil, err
+		}
+		tables[requestPayload.PredictionTable.Name] = predictionTable
```

Comment:
What if `requestPayload.PredictionTable.Name` is empty? Did we validate it?

Reply:
previously no, but I just added it